sat/cli/bootprep/input/image.py

```diff
@@ -479,3 +497,146 @@ def completed_successfully(self):
         return self.image_create_success and self.image_configure_success
 
     begin = begin_image_create
+
+
+class IMSInputImage(BaseInputImage, ABC):
```

Comment:
nit: FWIW, I don't think putting `ABC` as a superclass is necessary here, since it's a superclass of `BaseInputImage` too.

`abc.ABC` itself is just a trivial class with the metaclass of `ABCMeta`. See the source: https://github.com/python/cpython/blob/main/Lib/abc.py#L184

In addition, the metaclass is determined by the figuring out a subtype of the metaclasses of the bases and the explicitly specified metaclass. There is some (possibly mind-melting) information here, which is kind of interesting in its own right: https://docs.python.org/3/reference/datamodel.html#determining-the-appropriate-metaclass

Reply:
Before I added the `ims_data` abstract property to this class, I was getting a warning from PyCharm that said "Class IMSInputImage must implement all abstract methods". This warning is no longer present if I remove the `ABC` parent class now that I have a method on the class with the `abstractmethod` decorator. If I remove that abstract method, the warning comes back. I wonder if this is just a PyCharm limitation.